veggieseasons/lib/screens/settings.dart

```diff
@@ -3,19 +3,199 @@
 // found in the LICENSE file.
 
 import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
 import 'package:flutter/widgets.dart';
+import 'package:scoped_model/scoped_model.dart';
+import 'package:veggieseasons/data/preferences.dart';
+import 'package:veggieseasons/data/veggie.dart';
 import 'package:veggieseasons/styles.dart';
+import 'package:veggieseasons/widgets/settings_group.dart';
+import 'package:veggieseasons/widgets/settings_item.dart';
+
+class VeggieCategorySettingsScreen extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    final model = ScopedModel.of<Preferences>(context, rebuildOnChange: true);
+    final currentPrefs = model.preferredCategories;
+
+    return CupertinoPageScaffold(
+      navigationBar: CupertinoNavigationBar(
+        middle: Text('Preferred Categories'),
+        previousPageTitle: 'Settings',
+      ),
+      backgroundColor: Styles.scaffoldBackground,
+      child: FutureBuilder<Set<VeggieCategory>>(
+        future: currentPrefs,
+        builder: (context, snapshot) {
+          final items = <SettingsItem>[];
+
+          for (final category in VeggieCategory.values) {
+            CupertinoSwitch toggle;
+
+            // It's possible that category data hasn't loaded from shared prefs
+            // yet, so display it if possible and fall back to disabled switches
+            // otherwise.
+            if (snapshot.hasData) {
+              toggle = CupertinoSwitch(
+                value: snapshot.data.contains(category),
+                onChanged: (value) {
+                  if (value) {
+                    model.addPreferredCategory(category);
+                  } else {
+                    model.removePreferredCategory(category);
+                  }
+                },
+              );
+            } else {
+              toggle = CupertinoSwitch(
+                value: false,
+                onChanged: (value) {},
```

Comment:
This is not disabled though right?

Reply:
No. It should be `onChanged: null`.

Out of curiosity, if null is an acceptable value, why are the onChanged callbacks for switches `@required` (both here and in material.dart)?

Reply:
Just to make it explicit